resources/web/streaming_viewer/setup_viewer.js

```diff
@@ -26,32 +25,43 @@ function init() {
   ipInput = document.getElementById('IPInput');
   connectButton = document.getElementById('ConnectButton');
   modeSelect = document.getElementById('mode');
-  broadcast = document.getElementById('broadcast')
+  broadcast = document.getElementById('broadcast');
+
+  connectButton.onclick = connect;
 }
 
 function connect() {
   // This `streaming viewer` setups a broadcast streaming where the simulation is shown but it is not possible to control it.
   // For any other use, please refer to the documentation:
   // https://www.cyberbotics.com/doc/guide/web-simulation#how-to-embed-a-web-scene-in-your-website
   let playerDiv = document.getElementById('playerDiv');
-  view = new webots.View(playerDiv, mobileDevice);
+  if (!view)
+    view = new webots.View(playerDiv, mobileDevice);
   view.broadcast = broadcast.checked;
   view.setTimeout(-1); // disable timeout that stops the simulation after a given time
   const streamingMode = modeSelect.options[modeSelect.selectedIndex].value;
+
+  view.onready = function() {
```

Comment:
Maybe it is a good practice to use arrow functions when `this` of the new function is not needed.
```suggestion
  view.onready = _ => {
```

Reply:
thanks, fixed in 8d9e151